recipes/qpdf/all/patches/0001-libqpdf-cmake-deps-jpeg-zlib.patch

```diff
@@ -0,0 +1,50 @@
+diff --git a/libqpdf/CMakeLists.txt b/libqpdf/CMakeLists.txt
+index 7053e205..877c6f44 100644
+--- a/libqpdf/CMakeLists.txt
++++ b/libqpdf/CMakeLists.txt
+@@ -403,6 +403,9 @@ endif()
+ # use PIC for the object library so we don't have to compile twice.
+ set(OBJECT_LIB libqpdf_object)
+ add_library(${OBJECT_LIB} OBJECT ${libqpdf_SOURCES})
++find_package(JPEG REQUIRED)
++find_package(ZLIB REQUIRED)
++
+ if(OBJECT_LIB_IS_PIC)
+   target_compile_definitions(${OBJECT_LIB} PRIVATE libqpdf_EXPORTS)
+ endif()
+@@ -417,9 +420,9 @@ target_include_directories(${OBJECT_LIB}
+   ${qpdf_SOURCE_DIR}/libqpdf
+   ${CMAKE_CURRENT_BINARY_DIR})
+ target_link_directories(${OBJECT_LIB} INTERFACE ${dep_link_directories})
+-target_link_libraries(${OBJECT_LIB} INTERFACE ${dep_link_libraries})
++target_link_libraries(${OBJECT_LIB} INTERFACE ${dep_link_libraries} PUBLIC JPEG::JPEG ZLIB::ZLIB)
+ if(ATOMIC_LIBRARY)
+-  target_link_libraries(${OBJECT_LIB} INTERFACE ${ATOMIC_LIBRARY})
++  target_link_libraries(${OBJECT_LIB} INTERFACE ${ATOMIC_LIBRARY} PUBLIC JPEG::JPEG ZLIB::ZLIB)
+ endif()
+ 
+ set(LD_VERSION_FLAGS "")
+@@ -498,10 +501,8 @@ if(BUILD_SHARED_LIBS)
+     PUBLIC
+     $<BUILD_INTERFACE:${qpdf_INCLUDE}>
+     $<INSTALL_INTERFACE:include>)
+-  target_link_directories(${SHARED_LIB} PRIVATE ${dep_link_directories})
+-  target_link_libraries(${SHARED_LIB} PRIVATE ${dep_link_libraries})
+   if(ATOMIC_LIBRARY)
+-    target_link_libraries(${SHARED_LIB} PRIVATE ${ATOMIC_LIBRARY})
```

Comment:
Do you need to leave this line unchanged and put the `target_link_libraries` for JPEG / ZLIB outside the `if(ATOMIC_LIBRARY)` check?

Reply:
I just confirmed that `ATOMIC_LIBRARY` will be false on macOS. From the console output in the macOS logs, you can see that `ATOMIC_WORKED0` is a success:

```
-- Performing Test ATOMIC_WORKED0
-- Performing Test ATOMIC_WORKED0 - Success
``

`qpdf` only sets `ATOMIC_LIBRARY` when `ATOMIC_WORKED0` fails and `ATOMIC_WORKED1` succeeds.
Since it isn't set, JPEG and ZLIB are not linked in.

Reply:
Oh good point. I hope my new patch-set is more versatile, since i set now the dependencies in an additional target_link_libraries call outside of the conditional checks for every library type (object, static, shared)